onlinejudge/_implementation/command/test.py

```diff
@@ -47,12 +47,40 @@ def f(x):
     return True
 
 
-def compare_and_report(proc: subprocess.Popen, answer: str, elapsed: float, memory: Optional[float], test_input_path: pathlib.Path, test_output_path: Optional[pathlib.Path], *, mle: Optional[float], mode: str, error: Optional[float], does_print_input: bool, silent: bool, rstrip: bool) -> str:
+def compare_and_report(proc: subprocess.Popen, answer: str, elapsed: float, memory: Optional[float], test_input_path: pathlib.Path, test_output_path: Optional[pathlib.Path], *, mle: Optional[float], mode: str, error: Optional[float], does_print_input: bool, silent: bool, rstrip: bool, judge: Optional[str]) -> str:
+    rstrip_targets = ' \t\r\n\f\v\0'  # ruby's one, follow AnarchyGolf
+
     # prepare the comparing function
-    if error:  # float mode
+    if error is not None:  # float mode
         match = lambda a, b: compare_as_floats(a, b, error)
+    elif judge is not None:  # special judge mode
+
+        def match(a, b):
+            input = test_input_path.read_bytes()
+            # On Windows, a temp file is not created if we use "with" statement,
+            user_output = tempfile.NamedTemporaryFile(delete=False)
+            judge_result = False
+            try:
+                user_output.write(a.rstrip(rstrip_targets).encode())
+                user_output.seek(0)
+
+                arg0 = judge
+                arg1 = str(test_input_path.resolve())
+                arg2 = user_output.name
+                arg3 = str(test_output_path.resolve() or '')
+
+                actual_command = '{} {} {} {}'.format(arg0, arg1, arg2, arg3)
```

Comment:
```suggestion
                actual_command = '{} {} {} {}'.format(arg0, arg1, arg2, arg3)  # TODO: quote arguments for paths including spaces; see https://github.com/kmyk/online-judge-tools/pull/584
```

空白の問題はひとまず置いておくということなのでコメントを足しておきたい。コメントがなかったら単なるバグでも、あれば「ひとまず実装の見送りを選択しました」ということになって許される感じがある

Reply:
別の所にも書きましたが，コマンド部分のquote問題は`-c`指定の場合と同様なので，コマンド部分以外をquote処理して対応することにしました．

Reply:
👍

Reply:
よさそうに見えたが Windows でだめそう。つらい

Reply:
そもそもquoteしてない部分のログ出力が
```
[x] $ c:\python35\python.exe
```
みたいになっててバックスラッシュ足りてなさそうでやばそう
（恐らくfinally句でreturnしてて見えなかったバグが顕になっただけ）

Reply:
Windows 上なら `c:\python35\python.exe` は正しいパス (`/` の代わりに `\` が使われる) なので問題なさそう

Reply:
正しかった（すみません）